distributed/tests/test_scheduler.py

```diff
@@ -223,7 +223,7 @@ def random(**kwargs):
         secondary_worker_key_fractions = []
         for keys in x.__dask_keys__():
             # Iterate along rows of the array.
-            keys = {stringify(k) for k in keys}
+            keys = {k for k in keys}
```

Comment:
It looks like we should be able to drop this step.

Reply:
well, we should convert to set. I don't trust `__dask_keys__`